docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,349 @@
-*   `return!` executes another async workflow and returns its return value as a result.
+The `Async.Start` function is used to start the computation.
 
-Additionally, normal `let`, `use`, and `do` keywords can be used alongside the async versions just as they would in a normal function.
+Additionally, `Async.AwaitTask` is called in the `countFileBytes` function. This creates an `Async<'T>` that runs <xref:System.IO.File.WriteAllBytesAsync%2A> (which returns a `Task<byte[]>`) and returns its result.
 
-## How to start Async Code in F# #
+This is a key difference with the C#/VB style of `async` programming. In F#, asynchronous computations can be thought of as **Cold tasks**. They must be explicitly started to actually execute. This has some advantages, as it allows you to combine and sequence asynchronous work very easily.
 
-As mentioned earlier, async code is a specification of work to be done in another context which needs to be explicitly started. Here are two primary ways to accomplish this:
+## Combining asynchronous computations
 
-1.  `Async.RunSynchronously` will start an async workflow on another thread and await its result.
+Here is an example that builds upon the previous one by combining computations:
 
 ```fsharp
 open System
-open System.Net
+open System.IO
 
-let fetchHtmlAsync url = 
+let countFileBytes path =
     async {
-        let uri = Uri(url)
-        use webClient = new WebClient()
-        let! html = webClient.AsyncDownloadString(uri)
-        return html
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
     }
 
- // Execution will pause until fetchHtmlAsync finishes
- let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Parallel
+    |> Async.Ignore
+    |> Async.Start
 
- // you actually have the result from fetchHtmlAsync now!
- printfn "%s" html
- ```
+    Console.Read() |> ignore
+    0 // return an integer exit code
+```
+
+As you can see, the `main` function has quite a few more calls made. Conceptually, it does the following:
+
+1. Transform the command line arguments into `Async<unit>` computations with `Array.map`
+2. Create an `Async<'T[]>` that will schedule and run the `countFileBytes` computations in parallel when it is ran
+3. Create an `Async<unit>` that will run the parallel computation and ignore its result
+4. Explicitly run the last computation with `Async.Start`
 
-2.  `Async.Start` will start an async workflow on another thread, and will **not** await its result.
+When this program is ran, `countFileBytes`is ran in parallel for each command line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
```

Comment:
```suggestion
When this program runs, `countFileBytes` runs in parallel for each command line argument. Because asynchronous computations execute independently of program flow, there is no order in which they print their information and finish executing.
```